SwiftUIBasics/Views/SignUpView.swift

```diff
@@ -61,13 +49,31 @@ class SignUpViewModel: ObservableObject {
             .assign(to: \.isValidPasswordMatch, on: self)
             .store(in: &cancelableSet)
 
-        Publishers.CombineLatest4($isValidUsernameLength, $isValidPasswordLength, $isValidPasswordUpperCase, $isValidPasswordMatch)
-            .map { (a, b, c, d) in
-                return a && b && c && d
+        Publishers.CombineLatest3($isValidEmail, $isValidPassword, $isValidPasswordMatch)
+            .map { (isValidEmail, isValidPassword, isValidPasswordMatch) in
+                return isValidEmail && isValidPassword && isValidPasswordMatch
             }
             .assign(to: \.isValid, on: self)
             .store(in: &cancelableSet)
     }
+
+    private func isValidPassword(_ password: String) -> Bool {
```

Comment:
this sound like a good idea, but we need to tell the user what are the failing criteria for the password acceptance